### ISSUE TITLE
[iOS] incompatibility between Pushwoosh and FirebaseMessaging

### ISSUE THREAD

**Issue by matszafraniec** [Pushwoosh/pushwoosh-flutter#34]
Hello,

* Overall specs:
  * Pushwoosh package version: 2.0.2
  * Firebase messaging package version: 9.1.1
  * Flutter SDK version: 2.0.4
 
* Platform affected: **only iOS**.

Can you tell me if there can be some incompatibility when Pushwoosh package and FirebaseMessaging package are added in project?

After I've updated Flutter SDK to the recent version, on app startup* it throws me error as below:
```
Launching lib/main.dart on iPhoneMat in debug mode...
lib/main.dart:1

Automatically signing iOS for device deployment using specified development team in Xcode project: >>TEAM_ID<<
Xcode build done.                                           75.2s


(lldb) 2021-04-09 12:52:00.061033+0200 Runner[1031:139948] Pushwoosh: Initializing application runtime

7.3.0 - [Firebase/Core][I-COR000005] No app has been configured yet.
7.3.0 - [Firebase/Messaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:


https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging to ensure proper integration.


[PW] HWID: AEC8209D-76B7-4FBE-A68D-6944A4364B71
[PW] [I] -[Pushwoosh] Will show foreground notifications: 1
[PW] [I] -[PWPushNotificationsManagerCommon] Registered for push notifications: 7699d89c42be5e9162d2aea8807aecbded6a46023b8e772ac206dcab0bc56971

[PW] [I] -[PWRequestManager]
x
|    Pushwoosh request:

| Url:      https://EB148-1E466.api.pushwoosh.com/json/1.3/getInApps
| Payload:  {"request":{"device_type":1,"language":"en","application":"APP_CODE","userId":"AEC8209D-76B7-4FBE-A68D-6944A4364B71","v":"6.1.1","hwid":"AEC8209D-76B7-4FBE-A68D-6944A4364B71"}}


| Status:   "200 no error"
| Response: {"status_code":200,"status_message":"OK","response":{"inApps":[]}}

x

[PW] [I] -[PWPushNotificationsManagerCommon] Registered for push notifications: 7699d89c42be5e9162d2aea8807aecbded6a46023b8e772ac206dcab0bc56971
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x10)
    frame #0: 0x000000010549cc74 Runner`-[PushwooshPlugin onDidRegisterForRemoteNotificationsWithDeviceToken:](self=0x000000028325edf0, _cmd="onDidRegisterForRemoteNotificationsWithDeviceToken:", token=@"7699d89c42be5e9162d2aea8807aecbded6a46023b8e772ac206dcab0bc56971") at PushwooshPlugin.m:145:5142 	#pragma mark - PushNotificationDelegate
143
144 	- (void)onDidRegisterForRemoteNotificationsWithDeviceToken:(NSString *)token {
-> 145 	    _registerResult(token);
    	    ^
   146 	}
   147
   148 	- (void)onDidFailToRegisterForRemoteNotificationsWithError:(NSError *)error {

Target 0: (Runner) stopped.
Error connecting to the service protocol: failed to connect to http://127.0.0.1:50018/
Exited (sigterm)

```

'* please note that when I make fresh start (after uninstall and install the app again), it is working fine, but every time when I'm trying to open app again it throws an error.

**Comment by wfhm** [comment]
@matszafraniec Can you share a reproducer project so we could investigate it on our side?

**Comment by matszafraniec** [comment]
@wfhm 
Here you can find example project: https://drive.google.com/drive/folders/1dsYgOrnbzz2KZNtL5Vr1NGAhE6OrB8p3?usp=sharing

In main.dart file, there is initPushwooshService method. Please provide there your app_id and sender_id.

Please note that app will properly open after fresh install. Errors will be thrown when you try to launch the app again.

**Comment by matszafraniec** [comment]
@wfhm do you need any other informations from my side? Did you reproduce the issue?

**Comment by wfhm** [comment]
@matszafraniec,

I am sorry, not yet. The SDK team is currently overflown with tasks because of the new Google Play Store policies. I will let you know as soon as I get back to this task.

Meanwhile, I noticed that even you have added Firebase messaging dependency to your project, you did not implement any Firebase functionality in the Dart code. Could you please let me know what goal you want to finally achieve? The issue is most likely caused by both Pushwoosh and Firebase swizzling iOS registration callbacks, and, if you do not want your app to receive pushes from both Firebase and Pushwoosh, it might be a good idea to just try disabling Firebase swizzling:


> **FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:**
>
> **https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging**
>
> **to ensure proper integration.**

**Comment by matszafraniec** [comment]
Thank you for respond @wfhm and clarification.

I shared you only sample project, to give you entry point. In production project we use FirebaseMessaging for one of our functionality, we also using Pushwoosh to send push notifications for our users. So, for our needs both packages must work together.

**Comment by wfhm** [comment]
@matszafraniec,

Fixed in 2.0.3. The native registration method `[[UIApplication sharedApplication] registerForRemoteNotifications]` is called automatically when Firebase is initialized, and it triggers our Dart registration callbacks, but with a **nil** instead of the expected **FlutterResult** object.

**Comment by matszafraniec** [comment]
Thank you @wfhm . I can confirm that after update to 2.0.3, issue disappeared.
